File: frontend/app/api/player_api.py
from flask import Blueprint, request, jsonify
from flask_login import login_required
from app.models import db, Player

player_bp = Blueprint('player', __name__, url_prefix='/api/players')

@player_bp.route('', methods=['POST'])
@login_required
def add_player():
    """グループにプレイヤーを追加する"""
    data = request.json
    group_id = data.get('group_id')
    name = data.get('name')

    if not group_id or not name:
        return jsonify({'error': 'group_idとnameは必須です'}), 400

    player = Player(group_id=group_id, name=name)
    db.session.add(player)
    db.session.commit()

    return jsonify({'id': player.id, 'name': player.name}), 201

@player_bp.route('', methods=['GET'])
def get_players_by_group():
    """指定されたグループのプレイヤー一覧を返す"""
    group_id = request.args.get('group_id')
    if not group_id:
        return jsonify({'error': 'group_id is required'}), 400

    players = Player.query.filter_by(group_id=group_id).all()
    return jsonify([{'id': p.id, 'name': p.name} for p in players])

@player_bp.route('/<int:player_id>', methods=['GET'])
def get_player(player_id):
    """プレイヤー情報を1件取得"""
    player = Player.query.get_or_404(player_id)
    return jsonify({
        'id': player.id,
        'name': player.name,
        'group_id': player.group_id
    })
@player_bp.route('/<int:player_id>', methods=['DELETE'])
@login_required
def delete_player(player_id):
    """プレイヤーを削除（ただし大会参加中なら削除不可）"""
    player = Player.query.get_or_404(player_id)

    # そのプレイヤーが参加している大会があれば削除不可
    if player.tournament_participations:
        return jsonify({'error': 'このプレイヤーは大会に参加しているため削除できません'}), 400

    # プレイヤーを削除
    db.session.delete(player)
    db.session.commit()

    return jsonify({'message': 'プレイヤーを削除しました'}), 200

File: frontend/src/api/player_api.js
// mahjong/api/player_api.js
import { fetchWithAutoLogin } from '../utils/api_utils.js';
const API_BASE_URL = import.meta.env.VITE_API_BASE_URL;
const BASE_URL = `${API_BASE_URL}/api/players`;


/**
 * プレイヤー（参加者）を追加する
 * @param {Object} data - { group_id: number, name: string }
 * @returns {Promise<Object|null>}
 */
export async function createPlayer(data) {
  try {
    const response = await fetchWithAutoLogin(BASE_URL, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(data),
    });

    if (!response.ok) {
      console.warn('プレイヤー作成失敗:', await response.json());
      return null;
    }

    return await response.json();
  } catch (error) {
    console.error('プレイヤー作成通信エラー:', error);
    return null;
  }
}

/**
 * グループIDからプレイヤー一覧を取得
 * @param {number} groupId
 * @returns {Promise<Array>}
 */
export async function getPlayersByGroup(groupId) {
  try {
    const response = await fetchWithAutoLogin(`${BASE_URL}?group_id=${groupId}`);

    if (!response.ok) {
      console.warn('プレイヤー一覧取得失敗:', await response.json());
      return [];
    }

    return await response.json();
  } catch (error) {
    console.error('プレイヤー一覧取得通信エラー:', error);
    return [];
  }
}

/**
 * プレイヤーIDから詳細を取得
 * @param {number} playerId
 * @returns {Promise<Object|null>}
 */
export async function getPlayerById(playerId) {
  try {
    const response = await fetchWithAutoLogin(`${BASE_URL}/${playerId}`);

    if (!response.ok) {
      console.warn('プレイヤー詳細取得失敗:', await response.json());
      return null;
    }

    return await response.json();
  } catch (error) {
    console.error('プレイヤー詳細取得通信エラー:', error);
    return null;
  }
}

/**
 * プレイヤーを削除する（大会に参加している場合は削除不可）
 * @param {number} playerId - 削除対象のプレイヤーID
 * @returns {Promise<{ success: boolean, message: string }>}
 */
export async function deletePlayer(playerId) {
  try {
    const response = await fetchWithAutoLogin(`${BASE_URL}/${playerId}`, {
      method: 'DELETE',
    });

    if (!response.ok) {
      const error = await response.json();
      return { success: false, message: error.error || '削除に失敗しました' };
    }

    const result = await response.json();
    return { success: true, message: result.message };
  } catch (error) {
    console.error('プレイヤー削除エラー:', error);
    return { success: false, message: '通信エラーが発生しました' };
  }
}

